http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -461,4 +544,124 @@ private void addScopes(String scopes, Set<String> allScopes) {
+            }
+            jsonWebSignature.sign();
+            return jsonWebSignature;
+        }
+    }
+
+    private JsonWebEncryption encryptRequest(JsonWebSignature signedRequest) throws JoseException, IOException {
+        if (!deployment.getRequestObjectEncryptionAlgValuesSupported().contains(deployment.getRequestObjectEncryptionAlgValue())) {
+            throw log.invalidRequestObjectEncryptionAlgorithm();
+        } else if (!deployment.getRequestObjectEncryptionEncValuesSupported().contains(deployment.getRequestObjectEncryptionEncValue())) {
+            throw log.invalidRequestObjectEncryptionEncValue();
+        } else {
+            JsonWebEncryption jsonEncryption = new JsonWebEncryption();
+            jsonEncryption.setPayload(signedRequest.getCompactSerialization());
+            jsonEncryption.setAlgorithmConstraints(new AlgorithmConstraints(AlgorithmConstraints.ConstraintType.PERMIT, deployment.getRequestObjectEncryptionAlgValue(), deployment.getRequestObjectEncryptionEncValue()));
+            jsonEncryption.setAlgorithmHeaderValue(deployment.getRequestObjectEncryptionAlgValue());
+            jsonEncryption.setEncryptionMethodHeaderParameter(deployment.getRequestObjectEncryptionEncValue());
+            PublicKey encPublicKey = deployment.getEncryptionPublicKeyLocator().getPublicKey(null, deployment);
+            jsonEncryption.setKey(encPublicKey);
+            return jsonEncryption;
+        }
+    }
+
+    private String getRequestUri(String request) throws OidcException {
+        HttpPost parRequest = new HttpPost(deployment.getPushedAuthorizationRequestEndpoint());
+        List<NameValuePair> formParams = new ArrayList<NameValuePair>();
+        formParams.add(new BasicNameValuePair(REQUEST, request));
+        ClientCredentialsProviderUtils.setClientCredentials(deployment, parRequest, formParams);
+        parRequest.addHeader("Content-type", "application/x-www-form-urlencoded");
+
+        UrlEncodedFormEntity form = new UrlEncodedFormEntity(formParams, StandardCharsets.UTF_8);
+        parRequest.setEntity(form);
+
+        HttpResponse response;
+        try {
+            response = deployment.getClient().execute(parRequest);
+        } catch (Exception e) {
+            throw log.failedToSendPushedAuthorizationRequest(e);
+        }
+        if (response.getStatusLine().getStatusCode() != HttpStatus.SC_CREATED) {
+            EntityUtils.consumeQuietly(response.getEntity());
+            throw log.unexpectedResponseCodeFromOidcProvider(response.getStatusLine().getStatusCode());
+        }
+        try {
+            InputStream inputStream = response.getEntity().getContent();
+            StringBuilder textBuilder = new StringBuilder();
+            Reader reader = new BufferedReader(new InputStreamReader
+                    (inputStream, StandardCharsets.UTF_8));
+            int c = 0;
+            while ((c = reader.read()) != -1) {
+                textBuilder.append((char) c);
+            }
```

Comment:
We have a ServerRequest class that has some utility methods for sending requests to the OIDC provider. Do you think we could get a new utility method added there that this can make use of and can the new method make use of any existing utilities like readString from that class?